File: ProgramacionUR/Preparciales/Preparcial1/Punto1.cpp
#include<iostream>

using namespace std;

int main()
{
    int semilla = 5643;
    srand(semilla);
    
    int dado1 = 0; 
    int dado2 = 0; 
    int suma_dados = 0;
    int tiro_inicial = 0;
    char c;
    
    cout<<"Ingresa un caracter para tirar los dados"<<endl;
    cin>>c;
    dado1 = (rand()%6) + 1;
    dado2 = (rand()%6) + 1;
    suma_dados = dado1 + dado2;
    cout<<"Sacaste "<<suma_dados<<endl;
    tiro_inicial = suma_dados;
    
    if(suma_dados==2 || suma_dados==3 || suma_dados==12){
        cout<<"Perdiste!"<<endl;
    }else if(suma_dados==7 || suma_dados==11){
        cout<<"Ganaste!"<<endl;
    }else{
        cout<<"Fase de tiros secundarios"<<endl;
        cout<<"Tiro 1"<<endl;
        cout<<"Ingresa un caracter para tirar los dados"<<endl;
        cin>>c;
        dado1 = (rand()%6) + 1;
        dado2 = (rand()%6) + 1;
        suma_dados = dado1 + dado2;
        cout<<"Sacaste "<<suma_dados<<endl;
        
        if(suma_dados==7){
            cout<<"Perdiste!"<<endl;
        }else if(suma_dados==tiro_inicial){
            cout<<"Ganaste!"<<endl;
        }else{
            cout<<"Tiro 2"<<endl;
            cout<<"Ingresa un caracter para tirar los dados"<<endl;
            cin>>c;
            dado1 = (rand()%6) + 1;
            dado2 = (rand()%6) + 1;
            suma_dados = dado1 + dado2;
            cout<<"Sacaste "<<suma_dados<<endl;
            
            if(suma_dados==7){
                cout<<"Perdiste!"<<endl;
            }else if(suma_dados==tiro_inicial){
                cout<<"Ganaste!"<<endl;
            }else{
                cout<<"Tiro 3"<<endl;
                cout<<"Ingresa un caracter para tirar los dados"<<endl;
                cin>>c;
                dado1 = (rand()%6) + 1;
                dado2 = (rand()%6) + 1;
                suma_dados = dado1 + dado2;
                cout<<"Sacaste "<<suma_dados<<endl;
                
                if(suma_dados==7){
                    cout<<"Perdiste!"<<endl;
                }else if(suma_dados==tiro_inicial){
                    cout<<"Ganaste!"<<endl;
                }else{
                    cout<<"Tiro 4"<<endl;
                    cout<<"Ingresa un caracter para tirar los dados"<<endl;
                    cin>>c;
                    dado1 = (rand()%6) + 1;
                    dado2 = (rand()%6) + 1;
                    suma_dados = dado1 + dado2;
                    cout<<"Sacaste "<<suma_dados<<endl;
                    if(suma_dados==7){
                        cout<<"Perdiste!"<<endl;
                    }else if(suma_dados==tiro_inicial){
                        cout<<"Ganaste!"<<endl;
                    }else{
                        cout<<"Lanzaste 5 veces. Perdiste por w.";
                    }
                }
            }
        }
    }
    
    return 0;
}

File: ProgramacionUR/Preparciales/Preparcial1/Punto1.py
import random

semilla = 5472
random.seed(semilla)

dado1 = 0
dado2 = 0
suma_dados = 0
tiro_inicial = 0
c=''

print("Ingresa un caracter para tirar los dados")
c=input()
dado1=random.randint(1,6)
dado2=random.randint(1,6)
suma_dados=dado1+dado2;
tiro_inicial=suma_dados
print("Sacaste ",suma_dados)

if(suma_dados==2 or suma_dados==3 or suma_dados==12):
    print("Perdiste!")
elif(suma_dados==7):
    print("Ganaste!")
else:
    print("Fase de tiros secundarios")
    print("Tiro1")
    print("Ingresa un caracter para tirar los dados")
    c=input()
    dado1=random.randint(1,6)
    dado2=random.randint(1,6)
    suma_dados=dado1+dado2
    print("Sacaste ",suma_dados)
    if(suma_dados==7):
        print("Perdiste!")
    elif(suma_dados==tiro_inicial):
        print("Ganaste!")
    else:
        print("Tiro2")
        print("Ingresa un caracter para tirar los dados")
        c=input()
        dado1=random.randint(1,6)
        dado2=random.randint(1,6)
        suma_dados=dado1+dado2
        print("Sacaste ",suma_dados)
        if(suma_dados==7):
            print("Perdiste!")
        elif(suma_dados==tiro_inicial):
            print("Ganaste!")
        else:
            print("Tiro3")
            print("Ingresa un caracter para tirar los dados")
            c=input()
            dado1=random.randint(1,6)
            dado2=random.randint(1,6)
            suma_dados=dado1+dado2
            print("Sacaste ",suma_dados)
            
            if(suma_dados==7):
                print("Perdiste!")
            elif(suma_dados==tiro_inicial):
                print("Ganaste!")
            else:
                print("Tiro4")
                print("Ingresa un caracter para tirar los dados")
                c=input()
                dado1=random.randint(1,6)
                dado2=random.randint(1,6)
                suma_dados=dado1+dado2
                print("Sacaste ",suma_dados)
                
                if(suma_dados==7):
                    print("Perdiste!")
                elif(suma_dados==tiro_inicial):
                    print("Ganaste!")
                else:
                    print("Lanzaste 5 veces en total. Perdiste por w.")